.github/workflows/esp32-build.yaml

```diff
@@ -28,35 +28,45 @@ concurrency:
 jobs:
   esp-idf:
     runs-on: ubuntu-latest
-    container: espressif/idf:v${{ matrix.idf-version }}
+    container: espressif/idf:${{ matrix.idf-version }}
 
     strategy:
       fail-fast: false
 
       matrix:
+        esp-idf-target: ["esp32", "esp32c3"]
         idf-version:
-        - '4.4.6'
-        - '5.0.6'
-        - '5.1.3'
+         - 'v4.4.7'
+         - 'v5.0.6'
+         - 'v5.1.3'
+         - 'v5.2'
 
+        exclude:
+        - esp-idf-target: "esp32c3"
+          idf-version: 'v4.4.7'
+        - esp-idf-target: "esp32c3"
+          idf-version: 'v5.0.6'
+        - esp-idf-target: "esp32c3"
+          idf-version: 'v5.1.3'
     steps:
     - name: Checkout repo
-      uses: actions/checkout@v3
+      uses: actions/checkout@v4
 
-    - name: Build with idf.py
-      shell: bash
-      working-directory: ./src/platforms/esp32/
-      run: |
-        . $IDF_PATH/export.sh
-        idf.py reconfigure
-        idf.py build
-        idf.py size
-    - name: Print component size info with idf.py
-      shell: bash
-      working-directory: ./src/platforms/esp32/
-      run: |
-        . $IDF_PATH/export.sh
-        idf.py size-components
+    # - name: Build with idf.py
```

Comment:
Oops?

Reply:
on purpose, seems wasteful, unrelated? (or I'm missing something)

Reply:
but I'll leave it in then it can be cleaned up in another PR..